includes/admin/upgrades/migrations/migration-18_5.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @package WPBDP\Admin\Upgrades\Migrations
+ */
+
+/**
+ * Migration for DB version 18.5
+ */
+class WPBDP__Migrations__18_5 extends WPBDP__Migration {
+
+	/**
+	 * If payments were on, we disable all free plans and vice versa.
+	 * We scan all enabled plans first for this migration.
+	 * If payments was enabled, we disable the free plans.
+	 * If payments was disabled, we disabled the paid plans.
+	 *
+	 * @since x.x
+	 */
+	public function migrate() {
+		global $wpdb;
+		$payments_on = wpbdp_get_option( 'payments-on' );
+		$sql         = "SELECT `id`, `amount` FROM {$wpdb->prefix}wpbdp_plans WHERE `enabled` = %d";
+		$plans       = $wpdb->get_results( $wpdb->prepare( $sql, 1 ) );
+		$to_disable  = array();
+
+		foreach ( $plans as $plan ) {
+			if ( $payments_on && $plan->amount <= 0.0 ) {
+				$to_disable[] = $plan->id;
+			} elseif( ! $payments_on && $plan->amount > 0.0 ) {
+				$to_disable[] = $plan->id;
+			}
+		}
+
+		if ( ! empty( $to_disable ) ) {
+			$sql = "UPDATE {$wpdb->prefix}wpbdp_plans SET `enabled` = 0 WHERE `id` IN(" . implode( ', ', array_fill( 0, count( $to_disable ), '%d' ) ) . ")";
+			// Call $wpdb->prepare passing the values of the array as separate arguments.
+			$query = call_user_func_array( array( $wpdb, 'prepare' ), array_merge( array( $sql ), $to_disable ) );
```

Comment:
Can we use $wpdb->update here?

Reply:
`$wpdb->update` here would be too slow here for bulk updates and can take more time as opposed to a direct SQL query

Reply:
Looks like there's not a way to use "IN" in the update query. Marking this a resolved.